src/io/flutter/vmService/VMServiceManager.java

```diff
@@ -355,6 +367,12 @@ private void onFrameEventReceived() {
         addServiceExtension(extensionName);
       }
       pendingServiceExtensions.clear();
+
+      // Query for display refresh rate and add the value to the stream.
```

Comment:
it might have also been a deadlock because this code is within a synchronized block.
My rule of thumb is to avoid thinking of complex locking cases by never having code that performs any actions that might trigger a second lock while already inside one lock.
You could try moving this code after the end on the `synchronized (serviceExtensions)` block to see if that fixes the issue.

Reply:
hmm except I don't want `getDisplayRefreshRate()` called on every frame. We need to short circuit if `firstFrameEventReceived == true`.

Reply:
I tried a few variations of pulling the call outside the synchronized block, but all resulted in the same situation where we don't ever get a response. Leaving as is.

Reply:
sg. fine to leave as is. you can also set a boolean variable indicating where to short circuit outside the synchronization block.